clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -227,12 +227,42 @@ public boolean ready(Node node, long now) {
     }
 
     /**
+     * Diconnects the connection to a particular node, if there is one.
+     * Any pending ClientRequests for this connection will receive disconnections.
+     *
+     * @param nodeId The id of the node
+     */
+    @Override
+    public void disconnect(String nodeId) {
+        selector.close(nodeId);
```

Comment:
Have you considered implementing `Selector.disconnect` that would call `Selector.close(channel, true)` instead of keeping the state in `abortedSends`?